### PR TITLE
fix(base): shellcheck for modules.d/99base

### PR DESCRIPTION
**Pull request by haraldh**
Also remove some functions, which are not used and broken anyway.

### CODE REVIEW

File: modules.d/99base/dracut-lib.sh [hunk 1]
```diff
@@ -844,11 +836,11 @@ add_mount_point() {
 wait_for_mount() {
     local _name
     _name="$(str_replace "$1" '/' '\\x2f')"
-    printf '. /lib/dracut-lib.sh\nismounted "%s"\n' $1 \
+    printf '. /lib/dracut-lib.sh\nismounted "%s"\n' "$1" \
         >> "$hookdir/initqueue/finished/ismounted-${_name}.sh"
     {
-        printf 'ismounted "%s" || ' $1
-        printf 'warn "\"%s\" is not mounted"\n' $1
+        printf 'ismounted "%s" || ' "$1"
```
Comment (johannbg) [inline modules.d/99base/dracut-lib.sh:842]: Should not this be

`
printf ' "\"%s\" is mounted || ' "$1"
`
or
`
printf ' "\"%s\" already mounted || ' "$1"
`
to be inline with the following
`
printf 'warn "\"%s\" is not mounted"\n' "$1"
`
?

File: modules.d/99base/init.sh [hunk 2]
```diff
@@ -16,20 +16,18 @@ PATH=/usr/sbin:/usr/bin:/sbin:/bin
...
 # mount some important things
-[ ! -d /proc/self ] \
-    && mount -t proc -o nosuid,noexec,nodev proc /proc > /dev/null
-
-if [ "$?" != "0" ]; then
-    echo "Cannot mount proc on /proc! Compile the kernel with CONFIG_PROC_FS!"
-    exit 1
+if [ ! -d /proc/self ]; then
+    if ! mount -t proc -o nosuid,noexec,nodev proc /proc > /dev/null; then
+        echo "Cannot mount proc on /proc! Compile the kernel with CONFIG_PROC_FS!"
```
Comment (johannbg) [inline modules.d/99base/init.sh:21]: Probably a better wording here is 
`
echo "Cannot mount sysfs on /sys! Perhaps the kernel is not compiled with CONFIG_SYSFS?"
`
A subtle hint in a form of suggestion is better perceived than orders right.

**Review by johannbg — APPROVED** [review]

File: modules.d/99base/dracut-lib.sh (hunk 1 above)
Comment (haraldh) [reply modules.d/99base/dracut-lib.sh:842]: no.. inside `'` we don't have to quote `"`

File: modules.d/99base/init.sh (hunk 2 above)
Comment (haraldh) [reply modules.d/99base/init.sh:21]: created an issue: https://github.com/dracutdevs/dracut/issues/1260

**Pull request merged**